pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/SinkConfigUtils.java

```diff
@@ -399,8 +404,7 @@ public static SinkConfig convertFromDetails(FunctionDetails functionDetails) {
     }
 
     public static ExtractedSinkDetails validateAndExtractDetails(SinkConfig sinkConfig,
-                                                                 ClassLoader sinkClassLoader,
-                                                                 ClassLoader functionClassLoader,
```

Comment:
@nicoloboschi , In apache we are not validating functionClassLoader, 
FunctionClassLoader and functionClass name is used to resolve typeArg and inputClassLoader, but in apache, we are using typeArg is defined and we are using inputFunction.getTypePool() instead of inputClassLoader at the validation points.

Can you have a look if removing the param breaks any functionality?

Reply:
This difference comes from the fact that LS 2.10 has PIP-193 in it while AP 2.10 hasn't (it's only available since 3.0.0) 

We should apply the same fix to both the functions (both the sink and the pre-processing)
In this case the `sinkClassLoader` is the sink and `functionClassLoader ` is the preprocessing function